bootstrap/state-builder/src/build_balances.rs

```diff
@@ -50,12 +50,26 @@ fn main() {
 
         let amount: u64 = parts[3].parse().unwrap();
         let address_str = parts[5];
+        let script = parts[6];
 
         if i % 100_000 == 0 {
             println!("Processed {} UTXOs", i);
         }
 
-        if let Ok(address) = BitcoinAddress::from_str(address_str) {
+        // Load the address. The UTXO dump tool we use doesn't output all the addresses
+        // we support, so if parsing the address itself fails, we try parsing the script directly.
+        let address = if let Ok(address) = BitcoinAddress::from_str(address_str) {
+            Some(address)
+        } else if let Some(address) = BitcoinAddress::from_script(
+            &Script::from(hex::decode(script).expect("script must be valid hex")),
+            args.network.into(),
+        ) {
+            Some(address)
+        } else {
+            None
+        };
```

Comment:
Should we maybe move this to a common location so that we reuse it in the two files?

Reply:
That would be preferable, of course. In the interest of time, however, and given this is mostly one-off code, I'll keep it as-is and we can revisit it later if need be.